src/RandomSeed.cpp

```diff
@@ -0,0 +1,42 @@
+#include "global.h"
+#include "RandomSeed.h"
+#include <random>
+
+int GetRandomInt()
+{
+	static std::random_device rd;
+	static std::mt19937 gen(rd());
+	return std::uniform_int_distribution<int>(0, INT_MAX)(gen);
+}
+
+float GetRandomFloat()
+{
+	std::mt19937 gen(GetRandomInt());
```

Comment:
I think the body of this should be the same as above (use a random_device, instead of relying on GetRandomInt).